packages/webapp/src/members/components/withdraw-funds/withdraw-modal-step-form.tsx

```diff
@@ -0,0 +1,150 @@
+
+import { tokenConfig } from "config";
+import { SetValuesEvent, useUALAccount } from "_app";
+import { Asset, assetToLocaleString, assetToString } from "_app/utils";
+import { Button, Form, Heading, Text } from "_app/ui";
+
+import { WithdrawFundsFormFields } from "./withdraw-modal";
+
+interface Props {
+    availableFunds?: Asset;
+    formState: [WithdrawFundsFormFields, SetValuesEvent];
+    onPreview: (e: React.FormEvent) => Promise<void>;
+    onCancel: () => void;
+}
+
+export const WithdrawModalStepForm = ({
+    availableFunds,
+    formState,
+    onPreview,
+    onCancel,
+}: Props) => {
+    const [ualAccount] = useUALAccount();
+    const [fields, setFields] = formState;
+
+    const onChangeFields = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setFields(e);
+    };
+
+    const validateAccountField = (e: React.FormEvent<HTMLInputElement>) => {
+        const target = e.target as HTMLInputElement;
+        if (target.validity.valueMissing) {
+            target.setCustomValidity("Enter an account name");
+        } else {
+            target.setCustomValidity("Invalid account name");
+        }
+    };
+
+    const validateAmountField = (e: React.FormEvent<HTMLInputElement>) => {
+        const target = e.target as HTMLInputElement;
+        if (target.validity.rangeOverflow) {
+            target.setCustomValidity("Insufficient funds available");
+        } else {
+            target.setCustomValidity("Enter a valid withdrawal amount");
+        }
+    };
+
+    const clearErrorMessages = (e: React.FormEvent<HTMLInputElement>) => {
+        (e.target as HTMLInputElement).setCustomValidity("");
+    };
+
+    if (!availableFunds) return null;
+
+    const maxWithdrawal = Number(
+        assetToString(availableFunds!, availableFunds.precision).split(" ")[0]
+    );
+
+    const setMaxAmount = () =>
+        setFields({ target: { id: "amount", value: maxWithdrawal } });
+
+    const isThirdPartyWithdrawal = ualAccount.accountName !== formState[0].to;
+
+    const amountInputPreventChangeOnScroll = (
+        e: React.WheelEvent<HTMLInputElement>
+    ) => (e.target as HTMLInputElement).blur();
+
+    return (
+        <div className="space-y-4">
+            <Heading>Withdraw funds</Heading>
+            <Text>
+                Withdraw my funds from the Eden contract to a public{" "}
+                {tokenConfig.symbol} account.
+            </Text>
+            <Text>
+                Available:{" "}
+                <span className="font-medium">
+                    {assetToLocaleString(
+                        availableFunds,
+                        availableFunds.precision
+                    )}
+                </span>
+            </Text>
+            <form onSubmit={onPreview} className="space-y-3">
+                <Form.LabeledSet
+                    label={`${tokenConfig.symbol} account name (12 characters)`}
+                    htmlFor="to"
+                >
+                    <Form.Input
```

Comment:
suggestion: refactor this component to a `Form.ChainAccount` component. We could reuse it in the inductions!

Reply:
Done ✅